pyam/_aggregate.py

```diff
@@ -51,6 +53,39 @@ def _aggregate(df, variable, components=None, method=np.sum):
     return _group_and_agg(_df, [], method)
 
 
+def _aggregate_recursive(df, variable, components=None, method=np.sum):
+    _dfout = None
+    _df = df.copy()
+
+    mapping = []
```

Comment:
mapping usually implies a dictionary, so maybe call this `var_list` (and rename the `var_list` within the loop)?

Reply:
I decided to keep the `var_list` inside the loop, but `mapping` was renamed to `sub_variables`